platform/linux-generic/odp_ishm.c

```diff
@@ -2174,3 +2174,69 @@ void _odp_ishm_print(int block_index)
 
 	odp_spinlock_unlock(&ishm_tbl->lock);
 }
+
+int32_t odp_system_meminfo(odp_system_meminfo_t *info, odp_system_memblock_t memblock[],
+			   int32_t max_num)
+{
+	ishm_block_t *block;
+	int name_len, proc_index;
+	int32_t i;
+	uintptr_t addr;
+	uint64_t len, lost, page_size;
+	uint64_t lost_total = 0;
+	uint64_t len_total = 0;
+	int32_t num = 0;
+	const uint64_t huge_sz = odp_sys_huge_page_size();
+	const uint64_t normal_sz = odp_sys_page_size();
+
+	odp_spinlock_lock(&ishm_tbl->lock);
+	procsync();
+
+	for (i = 0; i < ISHM_MAX_NB_BLOCKS; i++) {
+		block = &ishm_tbl->block[i];
+
+		len = block->len;
+		if (len == 0)
+			continue;
+
+		lost = len - block->user_len;
+
+		if (num < max_num) {
+			name_len = strlen(block->name);
+			if (name_len >= ODP_SYSTEM_MEMBLOCK_NAME_LEN)
+				name_len = ODP_SYSTEM_MEMBLOCK_NAME_LEN - 1;
+
+			memcpy(memblock[num].name, block->name, name_len);
+			memblock[num].name[name_len] = 0;
```

Comment:
This is a performance optimization. As strncpy() writes always "num" bytes, I don't want to use ODP_SYSTEM_MEMBLOCK_NAME_LEN as "num". So, I check source string length before the call, and when that is known I can use as well memcpy(), since strncpy() would just check source string length again.

Reply:
OK. In that case, the `memcpy()` with the shorter len constant (ishm name vs. memblock name) could be faster than using `name_len` (enables compile-time optimizations).

Reply:
ISHM_NAME_MAXLEN is 128B and ODP_SYSTEM_MEMBLOCK_NAME_LEN is 64B. I'm assuming that names are much shorter in average (<20B). So, I'd prefer memcpy() of ~20B vs always 64B.